calico-vpp-agent/cni/cni_server.go

```diff
@@ -190,6 +242,23 @@ func (s *Server) Add(ctx context.Context, request *pb.AddRequest) (*pb.AddReply,
 		podSpec = &existingSpec
 	}
 
+	if podSpec.NetworkName != "" {
+		s.log.Infof("check overlapping networks")
+		overlap, err := s.checkOverlappingNetworks(podSpec)
```

Comment:
After the discussion last week, I think the behaviour we want in case of overlapping networks is not to prevent the creation of the interface, but just to drop the configuration of the route to the network prefix if it causes issues.